app/uk/gov/hmrc/helptosave/actors/EligibilityStatsActor.scala

```diff
@@ -61,12 +63,19 @@ class EligibilityStatsActor(scheduler:              Scheduler,
   def updateMetrics(table: Table): Unit = {
       def replaceSpaces(s: String) = s.replaceAllLiterally(" ", "-")
 
-    statsTable.foreach{
+    table.foreach{
       case (reason, channels) ⇒
         channels.foreach {
           case (channel, _) ⇒
-            metrics.registerAccountStatsGauge(replaceSpaces(reason), replaceSpaces(channel),
-              () ⇒ statsTable.get(reason).flatMap(_.get(channel)).getOrElse(0))
+            if (!registeredStats.contains(reason → channel)) {
+              logger.info(s"Registering gauge for (reason, channel) = ($reason, $channel) ")
+              metrics.registerAccountStatsGauge(
+                replaceSpaces(reason), replaceSpaces(channel),
+                () ⇒ statsTable.get(reason).flatMap(_.get(channel)).getOrElse(0)
```

Comment:
here you are still using `statsTable` is this expected ?

Reply:
yes, the `statsTable` is supposed to be the single point of reference for the registered gauges to go to when working out what the counts should be